### PR TITLE
Update ClangFormat configuration file for ClangFormat 14.0.0

### PR DESCRIPTION
**Pull request by per1234**
The **ClangFormat** configuration file hosted in this repository was developed for use with [**ClangFormat** 11.0.1](https://releases.llvm.org/11.0.1/tools/clang/docs/ClangFormat.html). The version of ClangFormat now in use by the Arduino IDE 2.x is 14.0.0:

https://github.com/arduino/arduino-ide/blob/19c0334a91cf562cf63edf0496eae16165499f8f/arduino-ide-extension/package.json#L164-L166

Configuration keys were added, removed, or changed in type in the interim.

The configuration file and its maintenance infrastructure are hereby updated for use with [**ClangFormat** 14.0.0](https://releases.llvm.org/14.0.0/tools/clang/docs/ClangFormat.html).

## Functional changes

The goal is always be to preserve consistent formatter output through any changes that must be made to the file. Since Arduino IDE 2.x is still in a pre-release state, the output of most interest is that of the stable production Arduino IDE 1.x and the Arduino Web Editor, which use the [**Artistic Style**](http://astyle.sourceforge.net/) code formatter.

The **ClangFormat** configuration file was designed to reproduce the results of the **Artistic Style** tool under [the default configuration provided by the Arduino IDE 1.x installation](https://github.com/arduino/Arduino/blob/master/build/shared/lib/formatter.conf). Unfortunately, it emerged at that time that some changes were unavoidable due to differences in the two tools, most often that **ClangFormat** tends to enforce whichever specific formatting style was chosen, without an option to simply accept whatever formatting the user happened to write, as was the case with **Artistic Style**.

Some of the newly added keys either offer a more lenient behavior or else allow specific behavior that  aligns with the output from **Artistic Style**. This opportunity was taken to achieve a closer alignment of the outputs from the two tools. This resulted in some diffs in the "golden master" test data used to verify the output from **ClangFormat** is as expected under Arduino's configuration. However, these diffs are actually the previously unavoidable unwanted formatting differences being reverted.

## Schema validation failure

The `validate` job of the "Check ClangFormat Configuration" workflow is still failing. This is caused by an error in the JSON schema, so it should not be considered an indication of a defect in this pull request or cause to delay its merge.

I have submitted a fix for that error and the job will pass once that is accepted:
https://github.com/SchemaStore/schemastore/pull/2384

### CODE REVIEW

**Comment by umbynos** [comment]
The [validation](https://github.com/arduino/tooling-project-assets/runs/7578400414?check_suite_focus=true) check is expected to fail?

**Comment by per1234** [comment]
Hi @umbynos. Yes, it is expected. Please see the explanation in the "Schema validation failure" section of the PR message.

**Comment by per1234** [comment]
The fix for the error in the `.clang-format` JSON schema (https://github.com/SchemaStore/schemastore/pull/2384) has now been merged. I reran the workflow and we are all ✔️

File: .github/workflows/check-clang-format.yml [hunk 1]
```diff
@@ -94,11 +95,27 @@ jobs:
...
 
+      - name: Download yq
+        id: download-yq
+        uses: MrOctopus/download-asset-action@1.0
+        with:
+          repository: mikefarah/yq
+          asset: yq_linux_amd64.tar.gz
+          target: ${{ env.YQ_INSTALL_PATH }}
+
+      - name: Install ClangFormat
```
Comment (umbynos) [inline .github/workflows/check-clang-format.yml:111]:
```suggestion
      - name: Install yq
```
leftover from the copy paste

**Review by umbynos — APPROVED** [review]

File: .github/workflows/check-clang-format.yml (hunk 1 above)
Comment (per1234) [reply .github/workflows/check-clang-format.yml]: Thanks Umberto!

**Pull request merged**